README.md

```diff
@@ -1,27 +1,74 @@
-To develop IntelliJ IDEA, you can use either IntelliJ IDEA Community Edition or IntelliJ IDEA Ultimate not older than 2017.3. To build and run the code:
-* Run **getPlugins.sh** / **getPlugins.bat** from the project root directory to check out additional modules.
-* If this git repository is not on 'master' branch you need to checkout the same branches/tags in android and android/tools-base git repositories.
-* Open the project.
-* If an error notification about a missing required plugin (e.g. Kotlin) is shown enable or install that plugin.
-* Configure a JSDK named "**1.8**", pointing to an installation of JDK 1.8.
-  * Add <JDK_18_HOME>/lib/tools.jar to the set of "**1.8**" JARs.
-* Configure a JSDK named "**IDEA jdk**" (case sensitive), pointing to an installation of JDK 1.6.
-  * If you don't plan to edit modules which use "IDEA jdk" you can configure "IDEA jdk" to point to the same installation of JDK 1.8 as "1.8" JDK. 
-  * Unless you're running on a Mac with an Apple JDK, add <JDK_HOME>/lib/tools.jar to the set of "**IDEA jdk**" JARs.
-* If 'Maven Integration' plugin is disabled add path variable 'MAVEN_REPOSITORY' pointing to '<USER_HOME>/.m2/repository' directory (File | Settings | Appearance & Behavior | Path Variables).
-* Use Build | Build Project to build the code.
-
-To run the IDE from the built code
-* use the provided shared run configuration "**IDEA**".
-
-To run tests apply these setting to the **default** JUnit run configuration type:
-* Working dir: 
-  `<IDEA_HOME>\bin`
-* VM parameters: 
-  `-ea -Xmx128M -Djava.system.class.loader=com.intellij.util.lang.UrlClassLoader -Didea.config.path=../test-config -Didea.system.path=../test-system`
-  
-To build an installation packages, run `ant` command in `<IDEA_HOME>` directory. See build.xml file for details.  
-
-You can find other useful information at [http://www.jetbrains.org](http://www.jetbrains.org). [Contribute section](http://www.jetbrains.org/display/IJOS/Contribute) of that site describes how you can contribute to IntelliJ IDEA.
\ No newline at end of file
+These instructions will help you build IntelliJ IDEA Community Edition from source code, which is the basis for IntelliJ Platform development. The following conventions will be used to refer to
+directories on your machine:
+* `<USER_HOME>` is your home directory.
+* `<IDEA_HOME>` is the root directory for the IntelliJ source code.
+* `<JDK_16_HOME>` is the root directory for the 1.6 JDK, which is optional.
+* `<JDK_18_HOME>` is the root directory for the 1.8 JDK.
+
+## Getting IntelliJ IDEA Community Edition Source Code
+IntelliJ IDEA Community Edition source code is available from
+`github.com/JetBrains/intellij-community` by either cloning or
+downloading a zip file (based on a branch) into `<IDEA_HOME>`. The
+default is the *master* branch. 
+
+The master branch contains the source code for the most recent public release of IntelliJ IDEA. The branch names and build numbers for older
```

Comment:
> The master branch contains the source code for the most recent public release of IntelliJ IDEA. 

This isn't quite true. Public releases are created from stable branches which are created 5-6 weeks before new major release from 'master' branch (see [our internal docs](https://confluence.jetbrains.com/display/IDEA/IntelliJ+Products+Versioning%2C+Branching+and+Build+Numbering) for details). I would write something like 'The master branch contains the source code which will be used to create the next major version of IntelliJ IDEA' instead.

Reply:
Thanks for clarifying. Updated per your suggestion.